statsmodels/discrete/count_model.py

```diff
@@ -0,0 +1,468 @@
+        params : array-like
+            The parameters of the model.
+
+        Returns
+        -------
+        loglike : float
+            The log-likelihood function of the model evaluated at `params`.
+            See notes.
+
+        Notes
+        --------
+        .. math:: \\ln L=\\sum_{y_{i}=0}\\ln(w_{i}+(1-w_{i})*P_{main\\_model})+
+            \\sum_{y_{i}>0}(\\ln(1-w_{i})+L_{main\\_model})
+            where P - pdf of main model, L - loglike function of main model.
+
+        """
+        return np.sum(self.loglikeobs(params))
+
+    def loglikeobs(self, params):
+        """
+        Loglikelihood for observations of Generic Zero Inflated model
+
+        Parameters
+        ----------
+        params : array-like
+            The parameters of the model.
+
+        Returns
+        -------
+        loglike : ndarray (nobs,)
+            The log likelihood for each observation of the model evaluated
+            at `params`. See Notes
+
+        Notes
+        --------
+        .. math:: \\ln L=\\ln(w_{i}+(1-w_{i})*P_{main\\_model})+
+            \\ln(1-w_{i})+L_{main\\_model}
+            where P - pdf of main model, L - loglike function of main model.
+
+        for observations :math:`i=1,...,n`
+
+        """
+        params_infl = params[:self.k_inflate]
+        params_main = params[self.k_inflate:]
+
+        y = self.endog
+        w = self.model_infl.predict(params_infl)
+        w[w >= 1.] = np.nextafter(1, 0)
+        llf_main = self.model_main.loglikeobs(params_main)
+        zero_idx = np.nonzero(y == 0)[0]
+        nonzero_idx = np.nonzero(y)[0]
+
+        llf = np.zeros_like(y, dtype=np.float64)
+        llf[zero_idx] = (np.log(w[zero_idx] +
+            (1 - w[zero_idx]) * np.exp(llf_main[zero_idx])))
+        llf[nonzero_idx] = np.log(1 - w[nonzero_idx]) + llf_main[nonzero_idx]
+
+        return llf
+
+    def fit(self, start_params=None, method='bfgs', maxiter=35,
+            full_output=1, disp=1, callback=None,
+            cov_type='nonrobust', cov_kwds=None, use_t=None, **kwargs):
+        if start_params is None:
+            offset = getattr(self, "offset", 0) + getattr(self, "exposure", 0)
+            if np.size(offset) == 1 and offset == 0:
+                offset = None
+            mod_poi = Poisson(self.endog, self.exog, offset=offset)
+            start_params = mod_poi.fit(disp=0).params
+            start_params = np.append(np.zeros(self.k_inflate), start_params)
+        mlefit = super(GenericZeroInflated, self).fit(start_params=start_params,
+                       maxiter=maxiter, disp=disp,
+                       full_output=full_output, callback=lambda x:x,
+                       **kwargs)
+
+        zipfit = self.result(self, mlefit._results)
+        result = self.result_wrapper(zipfit)
+
+        if cov_kwds is None:
+            cov_kwds = {}
+
+        result._get_robustcov_results(cov_type=cov_type,
+                                      use_self=True, use_t=use_t, **cov_kwds)
```

Comment:
where is this copied from? I think this should be automatically handled when creating the results instance.